### PR TITLE
Use owasp-java-html-sanitizer through antisamy-markup-sanitizer-plugin

### PR DESCRIPTION
**Pull request by raul-arabaolaza**
That library depends on Guava which is a library provided by core, by not using it directly but instead via this plugin extra guanrantees for compatibility are achieved. Because the probabilities of antisamy-markup-formatter plugin to detect some sort of incompatibility between owasp-java-sanitizer and jenkins core are bigger as is related to the main functionality provided.

I have also added a small test to smoke the HTML sanitization, so in the future, compatibility can be tested by using `-Djenkins.version` 

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by raul-arabaolaza** [comment]
Need to allow incrementals consumption to fix the build

**Comment by raul-arabaolaza** [comment]
@ovasilko @justnoxx If you have some time, please.

File: .mvn/maven.config [hunk 1]
```diff
@@ -0,0 +1 @@
+-Pconsume-incrementals
```
Comment (jtnord) [inline .mvn/maven.config:1]: you should not need this for this PR?
if you do then the version you are depending on has not been released and then this should make this project unreleasable

File: src/test/java/org/jenkinsci/plugins/electricflow/HtmlSanitizerTest.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+
+public class HtmlSanitizerTest {
+
+    @Test
+    public void simpleSanitize() {
+        String sanitized = HtmlUtils.getHtmlPolicy().sanitize("<html><head><script>alert('paco');</script></head><body><h1>Hi</h1></body></html>");
+        Assert.assertNotNull("Sanitized string should not be null", sanitized);
+        Assert.assertEquals("The sanitized html is not the one expected", "Hi", sanitized);
```
Comment (jtnord) [inline src/test/java/org/jenkinsci/plugins/electricflow/HtmlSanitizerTest.java:13]: confused me for a short while:  `h1` is not allowed -> https://github.com/jenkinsci/electricflow-plugin/blob/df11b959d642ade6a86633741b6b8a1393f8a95f/src/main/java/org/jenkinsci/plugins/electricflow/ui/HtmlUtils.java#L11

**Review by jtnord — APPROVED** [review]

File: .mvn/maven.config (hunk 1 above)
Comment (alecharp) [reply .mvn/maven.config:1]: and I don't see the extension to use that.

Comment (raul-arabaolaza) [reply .mvn/maven.config]: Correct, fixed now

**Review by jtnord — APPROVED** [review]

**Comment by raul-arabaolaza** [comment]
@justnoxx  gentle ping

**Review by ovasilko — APPROVED** [review]

**Pull request merged**